core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2190,6 +2209,85 @@ private Optional<OutputDataSizeEstimateResult> getEstimatedOutputDataSize()
             return Optional.of(new OutputDataSizeEstimateResult(new OutputDataSizeEstimate(estimateBuilder.build()), OutputDataSizeEstimateStatus.ESTIMATED_BY_PROGRESS));
         }
 
+        private Optional<OutputDataSizeEstimateResult> getEstimatedMarkDistinctStageOutputDataSize(Function<StageId, StageExecution> stageExecutionLookup)
+        {
+            if (!markDistinctChainEstimationEnabled) {
+                return Optional.empty();
+            }
+
+            if (!isSingleMarkDistinctStage(this.getStageInfo().getPlan())) {
+                return Optional.empty();
+            }
+
+            if (!isNoMorePartitions()) {
+                return Optional.empty();
```

Comment:
Comment?

Reply:
This is actually not important - we will have it most of the time - but we are not making use of this piece of information